test_aodndata/acorn/test_acorn_dm.py

```diff
@@ -0,0 +1,54 @@
+    'IMOS_ACORN_RV_20161121T003000Z_GUI_FV00_radial.nc': 'IMOS/ACORN/radial/GUI/2016/11/21',
+    'IMOS_ACORN_RV_20171014T060000Z_LANC_FV00_radial.nc': 'IMOS/ACORN/radial/LANC/2017/10/14',
+    'IMOS_ACORN_RV_20180907T210500Z_LEI_FV00_radial.nc': 'IMOS/ACORN/radial/LEI/2018/09/07',
+    'IMOS_ACORN_RV_20190825T053500Z_NNB_FV00_radial.nc': 'IMOS/ACORN/radial/NNB/2019/08/25',
+    'IMOS_ACORN_RV_20200718T140000Z_NOCR_FV00_radial.nc': 'IMOS/ACORN/radial/NOCR/2020/07/18',
+    'IMOS_ACORN_RV_20210611T085000Z_RRK_FV00_radial.nc': 'IMOS/ACORN/radial/RRK/2021/06/11',
+    'IMOS_ACORN_RV_20220504T010000Z_SBRD_FV00_radial.nc': 'IMOS/ACORN/radial/SBRD/2022/05/04',
+    'IMOS_ACORN_RV_20230401T012000Z_TAN_FV00_radial.nc': 'IMOS/ACORN/radial/TAN/2023/04/01',
+    'IMOS_ACORN_V_20180910T010000Z_BONC_FV00_sea-state.nc': 'IMOS/ACORN/vector/BONC/2018/09/10',
+    'IMOS_ACORN_V_20140804T010000Z_TURQ_FV00_sea-state.nc': 'IMOS/ACORN/vector/TURQ/2014/08/04',
+    'IMOS_ACORN_RV_20120507T053500Z_CSP_FV01_radial.nc': 'IMOS/ACORN/radial_quality_controlled/CSP/2012/05/07',
+    'IMOS_ACORN_RV_20130408T142000Z_CWI_FV01_radial.nc': 'IMOS/ACORN/radial_quality_controlled/CWI/2013/04/08',
+    'IMOS_ACORN_RV_20140309T012500Z_FRE_FV01_radial.nc': 'IMOS/ACORN/radial_quality_controlled/FRE/2014/03/09',
+    'IMOS_ACORN_RV_20150210T001000Z_GUI_FV01_radial.nc': 'IMOS/ACORN/radial_quality_controlled/GUI/2015/02/10',
+    'IMOS_ACORN_RV_20160111T023500Z_LEI_FV01_radial.nc': 'IMOS/ACORN/radial_quality_controlled/LEI/2016/01/11',
+    'IMOS_ACORN_RV_20171212T044500Z_NNB_FV01_radial.nc': 'IMOS/ACORN/radial_quality_controlled/NNB/2017/12/12',
+    'IMOS_ACORN_RV_20181113T123000Z_RRK_FV01_radial.nc': 'IMOS/ACORN/radial_quality_controlled/RRK/2018/11/13',
+    'IMOS_ACORN_RV_20191014T192000Z_TAN_FV01_radial.nc': 'IMOS/ACORN/radial_quality_controlled/TAN/2019/10/14',
+    'IMOS_ACORN_W_20090808T023000Z_CBG_FV01_wavespec.nc': 'IMOS/ACORN/gridded_1h-avg-wave-spectra_QC/CBG/2009/08/08',
+    'IMOS_ACORN_W_20090801T023000Z_CBG_FV01_wavespec.nc': 'IMOS/ACORN/gridded_1h-avg-wave-spectra_QC/CBG/2009/08/01',
+    'IMOS_ACORN_MW_20110228T073000Z_CBG_FV01_windp.nc': 'IMOS/ACORN/gridded_1h-avg-wind-map_QC/CBG/2011/02/28',
+    'IMOS_ACORN_MW_20110921T063000Z_SAG_FV01_windp.nc': 'IMOS/ACORN/gridded_1h-avg-wind-map_QC/SAG/2011/09/21',
+    'IMOS_ACORN_W_20110921T053000Z_SAG_FV01_wavep.nc': 'IMOS/ACORN/gridded_1h-avg-wave-site-map_QC/SAG/2011/09/21',
+    'IMOS_ACORN_W_20110921T053000Z_CSP_FV01_wavep.nc': 'IMOS/ACORN/gridded_1h-avg-wave-station-map_QC/CSP/2011/09/21'
+}
+
+
+class TestAcornDmHandler(HandlerTestCase):
+    def setUp(self):
+        self.handler_class = AcornHandler
+        super(TestAcornDmHandler, self).setUp()
+
+    def test_various_path(self):
+        for nc_file in ACORN_DM_VARIOUS.keys():
+            dest_path = AcornHandler.dest_path(nc_file)
+            self.assertEqual(dest_path, os.path.join(ACORN_DM_VARIOUS[nc_file], nc_file))
```

Comment:
Test above is to check destination path is correct but we need to add a test on whether the file is failing regex or not. For example:
```bash
# unknown types
test_unknown_type() {
    local file="IMOS_ACORN_RV_20171212T044500Z_NNB_FV01_unknown.nc"
    local type=`get_type $file`
    assertEquals "type $file" "" "$type"
}

test_match_regex() {
    local good_files bad_files

    good_files="$good_files IMOS_ACORN_RV_20120507T053500Z_CSP_FV01_radial.nc"
    good_files="$good_files IMOS_ACORN_W_20090808T023000Z_CBG_FV01_wavespec.nc"
    good_files="$good_files IMOS_ACORN_MW_20110921T083000Z_SAG_FV01_windp.nc"
    good_files="$good_files IMOS_ACORN_W_20110921T053000Z_SAG_FV01_wavep.nc"
    good_files="$good_files IMOS_ACORN_W_20110921T053000Z_CSP_FV01_wavep.nc"
    good_files="$good_files IMOS_ACORN_V_20150301T013000Z_CBG_FV01_1-hour-avg.nc"

    bad_files="$bad_files IMOS_ACORN_V_20150301T013000Z_CBG_FV00_1-hour-avg.nc"
    bad_files="$bad_files IMOS_ACORN_RV_20171014T060000Z_LANC_FV00_radial.nc"
    bad_files="$bad_files IMOS_ACORN_RV_20161121T003000Z_GUI_FV00_radial.nc"
    bad_files="$bad_files IMOS_ACORN_V_20180910T010000Z_BONC_FV00_sea-state.nc"
    bad_files="$bad_files IMOS_ACON_RV_20171014T060000Z_LANC_FV00_radial.nc"
    bad_files="$bad_files IMOS_ACORN_RV_201611203000Z_GUI_FV00_radial.nc"
    bad_files="$bad_files IMOS_ACORN_V_20180910T010000Z_BONC_FV00_sea-se.nc"
    bad_files="$bad_files IMOS_ACORN_RV_20120507T053500Z_FV01_radial.nc"
    bad_files="$bad_files IMOS_ACORN_RV_20191014T192000Z_TAN_FV01_radial.nc.some_suffix"
    bad_files="$bad_files prefix.IMOS_ACORN_RV_20191014T192000Z_TAN_FV01_radial.nc"

    local file
    for file in $good_files; do
        regex_filter $file
        assertEquals "regex $file" 0 $?
    done

    local file
    for file in $bad_files; do
        regex_filter $file
        assertNotEquals "regex $file" 0 $?
    done
}
```

Reply:
yeap I need to add this

Reply:
any progress on this?

Reply:
yeap 
in test_fv00_in_dm and test_fv01_in_rt